packages/jest-haste-map/src/index.js

```diff
@@ -291,7 +290,24 @@ class HasteMap extends EventEmitter {
    * 1. read data from the cache or create an empty structure.
    */
   read(): InternalHasteMap {
-    return this._parse(fs.readFileSync(this._cachePath, 'utf8'));
+    if (v8.deserialize) {
+      // This may throw. `_buildFileMap` will catch it and create a new map.
+      const {version, hasteMap} = v8.deserialize(
+        fs.readFileSync(this._cachePath),
+      );
+      if (version !== process.versions.v8) {
+        throw new Error('jest-haste-map: v8 versions do not match.');
+      }
+      return hasteMap;
```

Comment:
Setting null prototype is not necessary with `v8.deserialize`?

Reply:
Ah, that doesn't meet my expectation:

```
v8.deserialize(v8.serialize(Object.create(null))).__proto__ === Object.create(null).__proto__  // false
```

fix incoming